fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/CTEProperty.java

```diff
@@ -0,0 +1,83 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.sql.optimizer.base;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.starrocks.sql.optimizer.Group;
+import com.starrocks.sql.optimizer.GroupExpression;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CTEProperty implements PhysicalProperty {
+    private final Set<Integer> cteIds;
+
+    public static final CTEProperty EMPTY = new CTEProperty(ImmutableSet.of());
+
+    private CTEProperty(Set<Integer> cteIds) {
+        this.cteIds = cteIds;
+    }
+
+    public CTEProperty() {
+        this.cteIds = Sets.newHashSet();
+    }
+
+    public CTEProperty(int cteId) {
+        cteIds = Sets.newHashSet(cteId);
+    }
+
+    public Set<Integer> getCteIds() {
+        return cteIds;
+    }
+
+    public CTEProperty removeCTE(int cteID) {
+        CTEProperty p = new CTEProperty();
+        p.getCteIds().addAll(this.cteIds);
+        p.getCteIds().removeIf(c -> c.equals(cteID));
+        return p;
+    }
+
+    public boolean isEmpty() {
+        return cteIds.isEmpty();
+    }
+
+    public void merge(CTEProperty other) {
+        this.cteIds.addAll(other.cteIds);
+    }
+
+    @Override
+    public boolean isSatisfy(PhysicalProperty other) {
+        return true;
```

Comment:
why always true?

Reply:
Never call the method, and CTE property will prune invalid plan directly, don't neet enforce